### PR TITLE
Allow Private SSH Git Urls using FluxConfig

### PR DESCRIPTION
**Pull request by dashkan**
*Issue #, if available:*
#3195 

*Description of changes:*
Allows private ssh git urls

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jaxesn** [comment]
Thanks for the submission! Well make sure someone reviews

**Comment by dashkan** [comment]
This patch fails to bootstrap gitops because it cannot validate the git ssh url. I need to figure out how to test this end to end. Any guidance is appreciated as this is holding me back from using gitops.

File: go.mod [hunk 1]
```diff
@@ -29,6 +29,7 @@ require (
 	github.com/stretchr/testify v1.7.0
 	github.com/tinkerbell/rufio v0.0.0-20220606134123-599b7401b5cc
 	github.com/tinkerbell/tink v0.6.1-0.20220509141453-30fe9e015575
+	github.com/whilp/git-urls v1.0.0
```
Comment (chrisdoherty4) [inline go.mod:32]: Thanks for the contribution. Are there any alternatives to this lib - it doesn't seem to be maintained? We could also consider copying the functions we need into our repository.

Comment (dashkan) [reply go.mod:32]: How about a regex to validate the ssh url?

**Comment by dashkan** [comment]
Private repo can be provided using the following url format ssh://git@github.com/ORG/repo.git

**Comment by danbudris** [comment]
> 

Hey there @dashkan 

Is there anything we could do to improve the docs to make sure that the URL format is apparent for EKS A users?

**Comment by dashkan** [comment]
> > 
> 
> Hey there @dashkan
> 
> Is there anything we could do to improve the docs to make sure that the URL format is apparent for EKS A users?

I think a lot of people will go to their repo click the copy button add prefix with ssh://, which is a valid format. I think it would help to put an example of expected ssh url format for public and private repos.

**Pull request closed without merging**